spec/features/recipes/show_spec.rb

```diff
@@ -0,0 +1,58 @@
+require "rails_helper"
+
+RSpec.describe "/recipes/:id Show Page" do
+  before(:each) do
+    @ingredient_1 = Ingredient.create!(name: "Ground Beef", cost: 2)
+    @ingredient_2 = Ingredient.create!(name: "Salt", cost: 4)
+    @ingredient_3 = Ingredient.create!(name: "Tomatoe Sauce", cost: 6)
+
+    @recipe_1 = Recipe.create!(name: "Spaghetti", complexity: 4, genre: "Italian")
+    @recipe_2 = Recipe.create!(name: "Hamburger", complexity: 1, genre: "American")
+
+    RecipeIngredient.create!(recipe: @recipe_1, ingredient: @ingredient_1)
+    RecipeIngredient.create!(recipe: @recipe_1, ingredient: @ingredient_2)
+    RecipeIngredient.create!(recipe: @recipe_1, ingredient: @ingredient_3)
+
+    RecipeIngredient.create!(recipe: @recipe_2, ingredient: @ingredient_1)
+    RecipeIngredient.create!(recipe: @recipe_2, ingredient: @ingredient_2)
+  end
+
+  it "displays the recipe's name, complexity and genre" do
+    visit "/recipes/#{@recipe_1.id}"
+
+    expect(page).to have_content(@recipe_1.name)
+    expect(page).to have_content("Complexity: #{@recipe_1.complexity}")
+    expect(page).to have_content("Genre: #{@recipe_1.genre}")
+
+    visit "/recipes/#{@recipe_2.id}"
```

Comment:
Good job visiting two different show pages and looking for specific content on both. One thing that could improve these tests is the inclusion of `.to_not` expect statements in addition of the `.to` normal ones. You could use this by visiting one page, expecting that content, and then expecting to _not_ see any other content that should belong on a different page. Essentially the "positive" and "negative" cases. A lot of times implementing these kinds of expect statements can open your eyes to the way your program is actually working, and not just the way you assume it to work. Luckily yours is all good here, but keep this in mind!

Reply:
Example: 
```ruby
visit "/recipes/#{@recipe_1.id}"

expect(page).to have_content(@recipe_1.name)
expect(page).to_not have_content(@recipe_2.name)
# and another line for recipe_2's ingredients that are different from recipe 1. 

```